thefuck/rules/goenv_no_such_command.py

```diff
@@ -0,0 +1,29 @@
+import re
+from thefuck.utils import cache, for_app, replace_argument, replace_command, which
+from thefuck.specific.devenv import env_available, COMMON_TYPOS
+from subprocess import PIPE, Popen
+
+enabled_by_default = env_available
+
+
+@for_app('goenv')
+def match(command):
+    return 'goenv: no such command' in command.output
+
+
+def get_commands():
+    proc = Popen(['goenv', 'commands'], stdout=PIPE)
+    return [line.decode('utf-8').strip() for line in proc.stdout.readlines()]
+
+
+if which('goenv'):
+    get_commands = cache(which('goenv'))(get_commands)
+
+
+@for_app('goenv')
```

Comment:
`for_app` is not meant to decorate `get_new_command`. Doing so will make such method return either `True` or —for most cases `False`.